### PR TITLE
Use UiDevice to change screen-orientation while testing

### PR DESCRIPTION
**Pull request by XinyueZ**
Using UiDevice can really simulate human action to orient screen instead coding on the Activity.

### CODE REVIEW

**Comment by XinyueZ** [comment]
The same error like #171 while building on CI.
Everything goes well on local.

`test_shouldBeWatered` needs some analyses to make better.  💘

File: build.gradle [hunk 1]
```diff
@@ -25,6 +25,7 @@ buildscript {
         // App dependencies
         constraintLayoutVersion = '2.0.0-alpha2'
         espressoVersion = '3.0.1'
+        uiautomator = "2.1.3"
```
[the remark this answers is not in the document]
Comment (XinyueZ) [reply build.gradle]: 👍

File: build.gradle [hunk 2]
```diff
@@ -37,6 +37,7 @@ buildscript {
         roomVersion = '1.1.1'
         runnerVersion = '1.0.1'
         supportLibraryVersion = '28.0.0-rc01'
+        uiautomator = "2.1.3"
```
Comment (tiembo) [inline build.gradle]: Thanks for moving this! Can you also add a `Version` suffix and change capitalization so it reads `uiAutomatorVersion`?

File: app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt [hunk 3]
```diff
@@ -52,13 +52,14 @@ class GardenActivityTest {
     @Test fun onRotate_NavigationStaysOpen() {
         clickOnHomeIconToOpenNavigationDrawer()
 
-        // Rotate device to landscape
-        activityTestRule.activity.requestedOrientation = SCREEN_ORIENTATION_LANDSCAPE
-        checkDrawerIsOpen()
+        // Rotate device to different orientations
```
Comment (tiembo) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt]: remove comment

File: app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt [hunk 4]
```diff
@@ -52,13 +52,14 @@ class GardenActivityTest {
...
+        with(UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())) {
+                setOrientationLeft()
```
Comment (tiembo) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt]: Add comment: Rotate device to landscape

File: app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt [hunk 5]
```diff
@@ -52,13 +52,14 @@ class GardenActivityTest {
...
+                checkDrawerIsOpen()
 
-        // Rotate device back to portrait
-        activityTestRule.activity.requestedOrientation = SCREEN_ORIENTATION_PORTRAIT
-        checkDrawerIsOpen()
+                setOrientationRight()
```
Comment (tiembo) [inline app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt]: Add comment: Rotate device back to portrait

File: build.gradle (hunk 2 above)
Comment (XinyueZ) [reply build.gradle]: 👍

File: app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt (hunk 4 above)
Comment (XinyueZ) [reply app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt]: 👍

File: app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt (hunk 5 above)
Comment (XinyueZ) [reply app/src/androidTest/java/com/google/samples/apps/sunflower/GardenActivityTest.kt]: 👍

**Pull request merged**